flow/workflows/cdc_flow.go

```diff
@@ -236,25 +222,50 @@ func processTableAdditions(
 				DropFlowStats:         val.DropMirrorStats,
 				SkipDestinationDrop:   val.SkipDestinationDrop,
 			}
-		} else if val.RequestedFlowState == protos.FlowStatus_STATUS_RESYNC {
-			logger.Info("resync requested during table additions")
+		case protos.FlowStatus_STATUS_RESYNC:
+			logger.Info("resync requested", slog.String("operation", op))
 			state.ActiveSignal = model.ResyncSignal
 			// since we are adding to TableMappings, multiple signals can lead to duplicates
 			// we should ContinueAsNew after the first signal in the selector, but just in case
 			cfg.Resync = true
 			cfg.DoInitialSnapshot = true
 			state.DropFlowInput = &protos.DropFlowInput{
 				// to be filled in just before ContinueAsNew
-				FlowJobName:           "",
-				FlowConnectionConfigs: nil,
+				FlowJobName:           cfg.FlowJobName,
+				FlowConnectionConfigs: cfg,
 				DropFlowStats:         val.DropMirrorStats,
 				SkipDestinationDrop:   val.SkipDestinationDrop,
 				Resync:                true,
 			}
-		} else if val.RequestedFlowState == protos.FlowStatus_STATUS_PAUSED {
-			logger.Info("pause requested during table additions, ignoring")
+		case protos.FlowStatus_STATUS_PAUSED:
+			logger.Info("pause requested while busy, ignoring for now", slog.String("operation", op))
 		}
-	})
+	}
+}
+
+func processTableAdditions(
+	ctx workflow.Context,
+	logger log.Logger,
+	cfg *protos.FlowConnectionConfigs,
+	state *CDCFlowWorkflowState,
+	mirrorNameSearch temporal.SearchAttributes,
+) error {
+	flowConfigUpdate := state.FlowConfigUpdate
+	if len(flowConfigUpdate.AdditionalTables) == 0 {
```

Comment:
nit: not related to PR, but we already check this before calling the function so could remove one or the other